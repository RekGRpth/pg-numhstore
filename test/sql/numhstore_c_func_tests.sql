SELECT array_count(ARRAY['de','de','us']);
SELECT array_count(ARRAY[1,5,9,5,10,5,5]);
SELECT hstore_add('a=>1,b=>2'::hstore,'b=>1,c=>2'::inthstore);
SELECT hstore_add('a=>1,b=>1000000000'::hstore,'b=>1,c=>2'::inthstore);
SELECT hstore_add('a=>1.3,b=>2.5'::hstore,'b=>1.2,c=>2.4'::floathstore);
SELECT hstore_add('a=>1,b=>2'::hstore,'b=>1,c=>2'::inthstore);
SELECT hstore_sub('a=>2,b=>3'::hstore,'b=>1,c=>2'::inthstore);
SELECT array_uniq(ARRAY[5,4,13,4,5,2]);

SELECT tracker_id, install_date, device_id, 
array_count( array_agg( CASE WHEN Date(created_at) - install_date  <= 30 THEN Date(created_at) - install_date ELSE NULL END ) ) as daily_session_cohort
FROM 
(
  SELECT 1, i, generate_series(('2012-01-01'::date +i)::timestamp,('2012-01-01'::date +i+100)::timestamp - '1 hour'::interval, '1 hour'), '2012-01-01'::date
	FROM  generate_series(1,100) i)sessions(tracker_id,device_id, created_at, install_date
)
WHERE device_id=30
GROUP BY tracker_id, install_date, device_id;

SELECT tracker_id, install_date, device_id, 
array_count( array_agg( CASE WHEN Date(created_at) - install_date  <= 30 THEN Date(created_at) - install_date ELSE NULL END )::text[] ) as daily_session_cohort
FROM 
(
  SELECT 1, i, generate_series(('2012-01-01'::date +i)::timestamp,('2012-01-01'::date +i+100)::timestamp - '1 hour'::interval, '1 hour'), '2012-01-01'::date
	FROM  generate_series(1,100) i)sessions(tracker_id,device_id, created_at, install_date
)
WHERE device_id=30
GROUP BY tracker_id, install_date, device_id;
