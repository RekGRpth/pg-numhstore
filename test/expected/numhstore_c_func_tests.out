SELECT array_count(ARRAY['de','de','us']);
     array_count      
----------------------
 "de"=>"2", "us"=>"1"
(1 row)

SELECT array_count(ARRAY[1,5,9,5,10,5,5]);
               array_count               
-----------------------------------------
 "1"=>"1", "5"=>"4", "9"=>"1", "10"=>"1"
(1 row)

SELECT hstore_add('a=>1,b=>2'::hstore,'b=>1,c=>2'::inthstore);
          hstore_add          
------------------------------
 "a"=>"1", "b"=>"3", "c"=>"2"
(1 row)

SELECT hstore_add('a=>1,b=>1000000000'::hstore,'b=>1,c=>2'::inthstore);
              hstore_add               
---------------------------------------
 "a"=>"1", "b"=>"1000000001", "c"=>"2"
(1 row)

SELECT hstore_add('a=>1.3,b=>2.5'::hstore,'b=>1.2,c=>2.4'::floathstore);
             hstore_add             
------------------------------------
 "a"=>"1.3", "b"=>"3.7", "c"=>"2.4"
(1 row)

SELECT hstore_add('a=>1,b=>2'::hstore,'b=>1,c=>2'::inthstore);
          hstore_add          
------------------------------
 "a"=>"1", "b"=>"3", "c"=>"2"
(1 row)

SELECT hstore_sub('a=>2,b=>3'::hstore,'b=>1,c=>2'::inthstore);
          hstore_sub           
-------------------------------
 "a"=>"2", "b"=>"2", "c"=>"-2"
(1 row)

SELECT array_uniq(ARRAY[5,4,13,4,5,2]);
 array_uniq 
------------
 {2,4,5,13}
(1 row)

SELECT tracker_id, install_date, device_id, 
array_count( array_agg( CASE WHEN Date(created_at) - install_date  <= 30 THEN Date(created_at) - install_date ELSE NULL END ) ) as daily_session_cohort
FROM 
(
  SELECT 1, i, generate_series(('2012-01-01'::date +i)::timestamp,('2012-01-01'::date +i+100)::timestamp - '1 hour'::interval, '1 hour'), '2012-01-01'::date
	FROM  generate_series(1,100) i)sessions(tracker_id,device_id, created_at, install_date
)
WHERE device_id=30
GROUP BY tracker_id, install_date, device_id
 tracker_id | install_date | device_id | daily_session_cohort 
------------+--------------+-----------+----------------------
          1 | 01-01-2012   |        30 | "30"=>"24"
(1 row)

SELECT tracker_id, install_date, device_id, 
array_count( array_agg( CASE WHEN Date(created_at) - install_date  <= 30 THEN Date(created_at) - install_date ELSE NULL END )::text[] ) as daily_session_cohort
FROM 
(
  SELECT 1, i, generate_series(('2012-01-01'::date +i)::timestamp,('2012-01-01'::date +i+100)::timestamp - '1 hour'::interval, '1 hour'), '2012-01-01'::date
	FROM  generate_series(1,100) i)sessions(tracker_id,device_id, created_at, install_date
)
WHERE device_id=30
GROUP BY tracker_id, install_date, device_id;
 tracker_id | install_date | device_id | daily_session_cohort 
------------+--------------+-----------+----------------------
          1 | 01-01-2012   |        30 | "30"=>"24"
(1 row)

